src/lib/storage/index/b_tree/b_tree_index.cpp

```diff
@@ -5,14 +5,20 @@
 
 namespace opossum {
 
+size_t BTreeIndex::estimate_memory_consumption(ChunkOffset row_count, ChunkOffset distinct_count,
+                                               uint32_t value_bytes) {
+  // ToDo(anyone): If you use this index in combination with the Tuning subsystem, you need to properly implement this.
+  Fail("BTreeIndex::estimate_memory_consumption() is not implemented yet");
+}
+
 BTreeIndex::BTreeIndex(const std::vector<std::shared_ptr<const BaseSegment>>& segments_to_index)
     : BaseIndex{get_index_type_of<BTreeIndex>()}, _indexed_segments(segments_to_index[0]) {
   Assert((segments_to_index.size() == 1), "BTreeIndex only works with a single segment.");
   _impl =
       make_shared_by_data_type<BaseBTreeIndexImpl, BTreeIndexImpl>(_indexed_segments->data_type(), _indexed_segments);
 }
 
-uint64_t BTreeIndex::memory_consumption() const { return _impl->memory_consumption(); }
+size_t BTreeIndex::_memory_consumption() const { return _impl->memory_consumption(); }
```

Comment:
Not part of this PR, but doesn't `BTreeIndexImpl<DataType>::memory_consumption()` ignore the heap size of `std::string`s?

Reply:
I haven't looked into cpp-btree's implementation, so don't know what `_btree.bytes_used()` is actually doing. Could be that it just takes `sizeof(DataType)`. hm =/

Reply:
Looks like it:

```
if (stats.leaf_nodes == 1 && stats.internal_nodes == 0) {
      return sizeof(*this) +
          sizeof(base_fields) + root()->max_count() * sizeof(value_type);
    }
```

For now, it's probably easiest to track the heap cost of the stored keys/values manually during construction of the index. I think we should get this in before we merge this PR, as we would otherwise report blatantly false values.